### PR TITLE
Thumbnail: Right click to upload

### LINKED ISSUE #751: Editor/Browser: Thumbnail
We have lost the ability to click to upload a thumbnail.

There should be two areas you can upload a new thumbnail:

- Entity details panel.
- Editor form form panel.

Currently the only way to upload a thumbnail is to drag and drop an image onto the Entity details panel. In the editor it's impossible.

- [ ] Right click the thumbnail `Upload thumbnail` with icon `add_photo_alternative`. 
- [ ] Drag and drop over Editor form form panel just like the Entity details panel.
- [ ] Delete thumbnail @martastain ?

image

_Drag and drop uploading on the Entity details panel._
https://github.com/user-attachments/assets/a5ec9ad9-5ccb-44a0-b63c-8c47da727c62

### Notes

1. You might be able to re-use the drag and drop component logic/styles from Entity details panel. `ThumbnailUploader`.
2. Right click and uploading logic should **not** be inside `Thumbnail` component.

### PR DESCRIPTION
**Pull request by flynput**
Adding thumbnails to the project entities can now be done  also from the Editor page by dragging & dropping on image on the details panel. 
Implemented also classic file upload method, which can be accessed via the thumbnail context menu: right click -> upload new thumbnail. Available in both Browse and Editor pages.

https://github.com/user-attachments/assets/7333a140-5a51-4d92-9167-2c57e795d68f

Closes #751

### CODE REVIEW

**Comment by flynput** [comment]
@Innders WIP: Refactoring to unify context menu & upload handlers into one provider which stays on top of the StackedThumbnail component

File: src/components/EntityThumbnailUploader/ThumbnailUploaderProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  inputRef,
+}: Props) => {
+
+  const [_, setFileUploadInProgress] = useState(false)
+  const [ctxMenuShow] = useCreateContext()
+  const resetFileUploadState = () => setFileUploadInProgress(false)
+
+  const ctxMenuItems = () => [
+    {
+      label: 'Upload New Thumbnail',
```
Comment (Innders) [inline src/components/EntityThumbnailUploader/ThumbnailUploaderProvider.tsx:30]: Only capilized first letter


```suggestion
      label: 'Upload new thumbnail',
```

File: src/pages/EditorPage/EditorPanel.jsx [hunk 2]
```diff
@@ -506,28 +510,53 @@ const EditorPanel = ({
...
-                  data-tooltip="Delete"
-                />
-                <Link to={`/projects/${projectName}/browser`}>
-                  <Button icon="visibility" disabled={noSelection} data-tooltip="View in Browser" />
-                </Link>
-              </>
-            }
-          />
+        <Container>
+          <EntityThumbnailUploader
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:514]: The dropzone should span the full height of the right side panel.

image

File: src/pages/EditorPage/EditorPanel.jsx [hunk 3]
```diff
@@ -506,28 +510,53 @@ const EditorPanel = ({
...
+
+              refetch()
+            }}
+          >
+            <EntityDetailsHeader
+              values={nodeIds.map((id) => nodes[id]?.data)}
+              entityType={entityType}
+              tools={
+                <>
+                  <Button
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:535]: These icons are now really big?

image

File: src/components/EntityThumbnailUploader/EntityThumbnailUploader.tsx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        throw new Error('File is not an image')
+      }
+
+      let promises = []
+      for (const entity of entities) {
+        const { id, entityType, projectName } = entity
+
+        if (!projectName) throw new Error('Project name is required')
+
+        const promise = axios.post(
```
Comment (Innders) [inline src/components/EntityThumbnailUploader/EntityThumbnailUploader.tsx:123]: It seems the uploading logic is in this file and in [ThumbnailUploader](https://github.com/ynput/ayon-frontend/blob/d8c86bb66d5bbad8eec451cd5d15c3ff5f89f7bc/src/components/ThumbnailUploader/ThumbnailUploader.tsx#L50)?

Any reason why it's duplicated, could be a custom hook be used to that it is written once and reused twice?

File: src/pages/EditorPage/EditorPanel.jsx [hunk 5]
```diff
@@ -506,28 +510,53 @@ const EditorPanel = ({
...
-            }
-          />
+        <Container>
+          <EntityThumbnailUploader
+            isCompact
+            entities={isFetchingEntitiesDetails ? entitiesToQuery : entityDetailsData}
+            projectName={projectName}
+            entityType={entityType}
+            onUploaded={operations => {
+              const firstOperation = operations[0]
```
Comment (Innders) [inline src/pages/EditorPage/EditorPanel.jsx:520]: Only one thumbnail gets updates, which makes sense in this code. Use a forEach loop to update all.

image

File: src/components/EntityThumbnailUploader/EntityThumbnailUploader.tsx [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+        {isDraggingFile && (
+          <ThumbnailUploader
+            onFinish={handleThumbnailUpload}
+            onDragLeave={() => setIsDraggingFile(false)}
+            onDragOver={(e) => e.preventDefault()}
+            className="thumbnail-uploader"
+            entities={entities}
+          />
+        )}
+        <input type="file" onChange={handleFileUpload} ref={inputRef} />
```
Comment (Innders) [inline src/components/EntityThumbnailUploader/EntityThumbnailUploader.tsx:169]: When uploading using the input, we should still see the uploading progress like in drag and drop. 


https://github.com/user-attachments/assets/286474b7-79e8-4178-a6bb-0f856a5f7bcb

**Review by Innders — CHANGES_REQUESTED** [review]
Almost there, just a small things to go and then it will be perfect.

File: src/components/EntityThumbnailUploader/ThumbnailUploaderProvider.tsx (hunk 1 above)
Comment (Innders) [reply src/components/EntityThumbnailUploader/ThumbnailUploaderProvider.tsx:30]: And also use the `add_photo_alternative` icon to match the dropzone icon.

**Comment by flynput** [comment]
https://github.com/user-attachments/assets/fd3518d0-bb17-438f-8c01-01da966cdc12

**Review by Innders — CHANGES_REQUESTED** [review]
Works really nice, great job 👍 

There is just one tiny thing I noticed. On the editor page you have to drag the file over the header (red box) to trigger to dropzone. The trigger should be the full height (green box). So you can drop files by dragging anywhere over the right side panel.

image

https://github.com/user-attachments/assets/bd691975-4e4a-4bc1-96a7-653b7db3e04b

**Comment by flynput** [comment]
> Works really nice, great job 👍
> 
> There is just one tiny thing I noticed. On the editor page you have to drag the file over the header (red box) to trigger to dropzone. The trigger should be the full height (green box). So you can drop files by dragging anywhere over the right side panel.
> 
> image
> 
>  Screen.Recording.2024-09-05.at.12.17.24.mov

Updated!

**Review by Innders — APPROVED** [review]
Works. Approved!

**Pull request merged**